bin/pytorch_inference/Main.cc

```diff
@@ -69,11 +69,18 @@ torch::Tensor infer(torch::jit::script::Module& module_,
         }
 
         auto output = module_.forward(inputs);
+        LOG_DEBUG(<< "output_" << i << ": " << output);
         if (output.isTuple()) {
             // For transformers the result tensor is the first element in a tuple.
             all.push_back(output.toTuple()->elements()[0].toTensor());
         } else {
-            all.push_back(output.toTensor());
+            auto outputTensor = output.toTensor();
+            auto sizes = outputTensor.sizes();
+            if (sizeof(sizes) > 1) {
```

Comment:
`sizeof` returns the size of the _thing_ in bytes, this conditional will always be true. Do we need this logic? I thought the specialisation of `writeInferenceResults()` was the fix here

Reply:
yea, I was just looking into these errors and this is definitely wrong. I was trying to check for the number of dimensions, but I checked the wrong thing. I'll double check, but I believe that this reshape is required to be able to do that cat (which was what was throwing the error initially). The change to `writeInferenceResults()` was also required further down.

Reply:
yea, I just double checked, without the reshape here we get the `zero-dimensional tensor (at position 0) cannot be concatenated` exception. But the if statement is wrong for the reason you mentioned.

Reply:
I've just changed it to do what I was trying to do (I think), reshape the tensor to be 1-dimensional if it was zero dimensional.

Reply:
I think you've got it now